Setup.pm

```diff
@@ -6,68 +6,133 @@ use strict;
 
 use lib q(.);
 use File::Path qw( mkpath );
 use File::Spec::Functions qw(catfile);
 use File::Basename qw(basename);
 use Config;
+use Cwd qw(cwd);
 
 use Win32;
 use Win32::API;
 use Win32::Shortcut;
 use Win32::TieRegistry;
 
-our $VERSION           = '0.01';
+BEGIN { Win32::Unicode::InternetShortcut->CoInitialize(); }
+END { Win32::Unicode::InternetShortcut->CoUninitialize(); }
+
+our $VERSION           = '0.02';
 my $SHCNE_ASSOCCHANGED = 0x8_000_000;
 my $SCNF_FLUSH         = 0x1000;
 
+my $ORGANIZATION = 'ActiveState';
+my $PROJECT      = 'perl-5.32';
+my $NAMESPACE    = "$ORGANIZATION/$PROJECT";
+my $PLATFORM_URL = "https://platform.activestate.com/$NAMESPACE";
+
 # Import Win32 function: `void SHChangeNotify(int eventId, int flags, IntPtr item1, IntPtr item2)`
 
 my $SHChangeNotify = Win32::API::More->new( 'shell32', 'SHChangeNotify', 'iiPP', 'V' );
 if ( not defined $SHChangeNotify ) {
     die "Can't import SHChangeNotify: ${^E}\n";
 }
 
-sub update_win32_shell() {
+sub update_win32_shell {
     $SHChangeNotify->Call( $SHCNE_ASSOCCHANGED, $SCNF_FLUSH, 0, 0 );
     return;
 }
 
-sub create_shortcut {
-    my $start_menu_path = Win32::GetFolderPath(Win32::CSIDL_STARTMENU());
-    my $start_menu_base = catfile($start_menu_path, 'ActiveState');
+sub desktop_dir_path {
+    return Win32::GetFolderPath(Win32::CSIDL_DESKTOPDIRECTORY());
+}
 
-    my $target  = $Config{perlpath};
-    my $icon    = q();
-    my $lnkPath = catfile($start_menu_base, 'Perl.lnk');
+sub start_menu_path {
+    return Win32::GetFolderPath(Win32::CSIDL_STARTMENU());
+}
 
-    mkpath($start_menu_base);
+sub create_internet_shortcut {
+    my $target   = shift;
+    my $icon     = shift;
+    my $linkPath = shift;
 
     if ( -e $lnkPath ) {
         unlink $lnkPath;
     }
 
-    #print "Creating shortcut: $lnkPath -> $target\n";
+    #print "Creating internet shortcut: $lnkPath -> $target\n";
+    my $link = Win32::Unicode::InternetShortcut->new;
+    $link->save($lnkPath, $target);
+    $link->load($lnkPath);
+
+    ($link->{target} eq $target) || die "Not the same url\n";
+
+    return;
+}
+
+sub create_shortcut {
+    my $target   = shift;
+    my $icon     = shift;
+    my $linkPath = shift;
+    my $location = shift;
+
+    if ( -e $lnkPath ) {
+        unlink $lnkPath;
+    }
+
+    #print "Creating application shortcut: $lnkPath -> $target\n";
     my $LINK = Win32::Shortcut->new();
     $LINK->{'Path'} = $target;
-
-    # $LINK->{'WorkingDirectory'} = '';
-    $LINK->{'IconLocation'} = $icon;
-    $LINK->{'IconNumber'}   = 0;
+    $LINK->{'IconLocation'}     = $icon;
+    $LINK->{'IconNumber'}       = 0;
+    $LINK->{'WorkingDirectory'} = $location;
     $LINK->Save($lnkPath);
     $LINK->Close();
 
     return;
 }
 
+sub create_internet_shortcuts {
+    my $target  = $PLATFORM_URL;
+    my $icon    = q();
+    my $lnkName = "$NAMESPACE Web.url";
+
+    my $start_menu_base = catfile(start_menu_path(), $ORGANIZATION);
+    mkpath($start_menu_base);
```

Comment:
One small idea here; `mkpath` returns the number of directories created, so checking that the return value is `1` could be interesting.

You'd probably want to wrap that in a test, though, because failing to create something that already exists should be fine.

  unless (-d $start_menu_base) {
    my $success = mkpath($start_menu_base);
    die "Couldn't create start menu shortcut path '$start_menu_base': $!" unless $success == 1;
  }

I don't feel strongly about this; it's a question of how defensive you want to get here.